spec/classes/init_spec.rb

```diff
@@ -740,43 +603,25 @@
             it { is_expected.to contain_class('consul').with_init_style('systemd') }
             it { is_expected.to contain_file('/etc/systemd/system/consul.service').with_content(%r{consul agent}) }
           end
-        elsif facts[:os]['release']['major'].to_i == 7
-          context 'On a redhat 7 based OS' do
+        elsif facts[:os]['release']['major'].to_i >= 7
+          context 'On a RedHat 7 and newer based OS' do
             it { is_expected.to contain_class('consul').with_init_style('systemd') }
             it { is_expected.to contain_file('/etc/systemd/system/consul.service').with_content(%r{consul agent}) }
           end
-        elsif facts[:os]['release']['major'].to_i == 20
-          context 'On a fedora 20 based OS' do
+        elsif facts[:os]['release']['major'].to_i >= 20
```

Comment:
Given the `>= 7` above, this will never match? (the tests are identical, so there is no issue as such, just weird code)

Reply:
fixed